plugins/pulp_docker/plugins/registry.py

```diff
@@ -194,34 +231,125 @@ def create_download_request(self, image_id, file_name, destination_dir):
         self.add_auth_header(req)
         return req
 
-    def add_auth_header(self, request):
+
+class V2Repository(object):
+    """
+    This class represents a Docker v2 repository.
+    """
+    API_VERSION_CHECK_PATH = '/v2/'
+    LAYER_PATH = '/v2/{name}/blobs/{tarsum}'
```

Comment:
On 08/03/2015 11:16 AM, Michael Hrivnak wrote:
>> +    LAYER_PATH = '/v2/{name}/blobs/{tarsum}'
> 
> I would replace |tarsum| with |digest|, since tarsum is just one kind of
> digest that could be used.

The "tarsum" language is actually the wording used in the upstream
dockumentation (pun intended) for the blobs. See:

https://github.com/docker/distribution/blob/release/2.0/docs/spec/api.md#pulling-a-layer

However, it seems that upstream is inconsistent with their language, as
they also use the word "digest" here:

https://github.com/docker/distribution/blob/release/2.0/docs/spec/api.md#get-blob

I'll go with digest so it's consistent with the Manifest model.

-- 
Randy Barlow